src/server/game/Entities/Player/Player.cpp

```diff
@@ -27243,3 +27247,186 @@ uint32 Player::DoRandomRoll(uint32 minimum, uint32 maximum)
 
     return roll;
 }
+
+uint32 Player::PlayScene(uint32 sceneId, Position const* position /*= nullptr*/)
+{
+    SceneTemplate const* sceneTemplate = sObjectMgr->GetSceneTemplate(sceneId);
+    return PlaySceneByTemplate(sceneTemplate);
+}
+
+uint32 Player::PlaySceneByTemplate(SceneTemplate const* sceneTemplate, Position const* position/* = nullptr*/)
+{
+    if (sceneTemplate == nullptr)
+        return 0;
+
+    SceneScriptPackageEntry const* entry = sSceneScriptPackageStore.LookupEntry(sceneTemplate->scenePackageId);
+
+    if (entry == nullptr)
+        return 0;
+
+    // By default, take player position
+    if (position == nullptr)
+    {
+        Position loc = GetPosition();
+        position = &loc;
```

Comment:
@runningnak3d you can't improve if you get offended when I suggest something.

@Traesh Pull Request review cannot replace proper code analysis, debugging and testing. I doubt any of the reviewer will run a static analysis tool (like Visual Studio /analyze or Clang Static Analyzer) or a dynamic analysis tool (like Memcheck for memory-related issues and Helgrind for thread/lock related issues), this work is in hand of the author of the Pull Request. It's a lot of work but it's the only way to ensure mistakes (which we all do) don't end up in the commit and get merged into master branch.